tests/charm++/atsync/atsync.C

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+#include <thread>
+
+#include "atsync.decl.h"
+
+/*readonly*/ CProxy_Main mainProxy;
+/*readonly*/ int nElements;
+/*readonly*/ int iters;
+
+/*mainchare*/
+class Main : public CBase_Main
+{
+public:
+  Main(CkArgMsg* m)
+  {
+    // Process command-line arguments
+    nElements = 5;
+    if (m->argc > 1)
+      nElements = atoi(m->argv[1]);
+    iters = 10;
+    if (m->argc > 2)
+      iters = atoi(m->argv[2]);
+    delete m;
+
+    // Start the computation
+    CkPrintf("Running atsync on %d processors for %d elements\n", CkNumPes(), nElements);
+
+    mainProxy = thisProxy;
+    CProxy_atsync arr = CProxy_atsync::ckNew();
+    for (int i = 0; i < nElements; i++)
+    {
+      arr[i].insert();
+    }
+    arr.doneInserting();
+
+    arr[0].start();
+  };
+
+  void done(void)
+  {
+    CkPrintf("All done\n");
+    CkExit();
+  };
+};
+
+class atsync : public CBase_atsync
```

Comment:
Why the weird style for a class/chare name?

Reply:
I was trying to match an existing example, but obviously this should be cleaned up, I'll do that.